src/parser/view_transform.c

```diff
@@ -4647,7 +4695,8 @@ mq_check_rewrite_select (PARSER_CONTEXT * parser, PT_NODE * select_statement)
 	    {
 	      from->next = mq_rewrite_vclass_spec_as_derived (parser, select_statement, from->next, NULL);
 	    }
-	  else if (from->next->info.spec.derived_table_type == PT_IS_SUBQUERY)
+	  else if (from->next->info.spec.derived_table_type == PT_IS_SUBQUERY
+		   || from->info.spec.derived_table_type == PT_DERIVED_DBLINK_TABLE)
```

Comment:
Is it not necessary to modify the code below on line 4726?

`if (is_union_translation == false && from && from->info.spec.derived_table_type == PT_IS_SUBQUERY)`

Reply:
Ok, I fixed it.